icaruscode/Decode/MCDecoderICARUSTPCwROI_module.cc

```diff
@@ -498,110 +499,54 @@ void MCDecoderICARUSTPCwROI::processSingleLabel(art::Event&
     // Require a valid handle
     if (digitVecHandle.isValid() && digitVecHandle->size()>0 )
     {
-        // Sadly, the RawDigits come to us in an unsorted condition which is not optimal for
-        // what we want to do here. So we make a vector of pointers to the input raw digits and sort them
-        std::vector<const raw::RawDigit*> rawDigitVec;
-
-        // Ugliness to fill the pointer vector...
-        for(size_t idx = 0; idx < digitVecHandle->size(); idx++) rawDigitVec.emplace_back(&digitVecHandle->at(idx)); //art::Ptr<raw::RawDigit>(digitVecHandle, idx).get());
-
-        // Sort (use a lambda to sort by channel id)
-        std::sort(rawDigitVec.begin(),rawDigitVec.end(),[](const raw::RawDigit* left, const raw::RawDigit* right) {return left->Channel() < right->Channel();});
-
-        // Declare a temporary digit holder and resize it if downsizing the waveform
-        unsigned int               dataSize = art::Ptr<raw::RawDigit>(digitVecHandle,0)->Samples(); //size of raw data vectors
-        raw::RawDigit::ADCvector_t rawDataVec(dataSize);
-
-        using BoardToChannelArrayPairMap = std::map<unsigned int, ChannelArrayPair>;
-
-        BoardToChannelArrayPairMap  boardToChannelArrayPairMap;
-        std::map<unsigned int, int> boardWireCountMap;
-        const unsigned int          MAXCHANNELS(64);
-
-        // Commence looping over raw digits
-        for(const auto& rawDigit : rawDigitVec)
-        {
-            raw::ChannelID_t channel = rawDigit->Channel();
-
-            ChannelToBoardWirePlaneMap::const_iterator channelToBoardItr = fChannelToBoardWirePlaneMap.find(channel);
-
-            if (channelToBoardItr == fChannelToBoardWirePlaneMap.end())
+        const unsigned int dataSize = art::Ptr<raw::RawDigit>(digitVecHandle,0)->Samples(); //size of raw data vectors
+
+	std::map<unsigned int, std::vector<const raw::RawDigit*> > boardToRawDigitMap;
+	for(size_t idx = 0; idx < digitVecHandle->size(); idx++) {
+	  const raw::RawDigit* rawDigit = &digitVecHandle->at(idx);
```

Comment:
I **suggest** that this loop be made simpler:
```suggestion
	for(const raw::RawDigit& rawDigit: *digitVecHandle) {
```
having `rawDigit` a reference. The two places where it is used below will need to be adapted.

Reply:
I think this will require to change more that two places, as the vector in the map will have to change type. Since this is how the code was written before, I think this is not in the scope of the current changes

Reply:
ah no, this is actually doable without changing the map type (which would be a nightmare)